fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/FedoraLdpIT.java

```diff
@@ -2476,6 +2503,41 @@ public void testContainerLastModified() throws Exception {
         }
     }
 
+    @Test
+    public void testConcurrentPutDatastreamResource() throws IOException, InterruptedException {
+        final String pid = getRandomUniqueId();
+        createObject(pid);
+
+        final String ds = "ds2";
+        final RequestThread t1 = new RequestThread(putDSFileMethod(pid, ds, TEST_FILE));
+        final RequestThread t2 = new RequestThread(putDSFileMethod(pid, ds, TEST_FILE));
+        t1.start();
+        Thread.sleep(100);
+        t2.start();
+        t1.join();
+        t2.join();
+
+        assertEquals(CREATED.getStatusCode(), t1.status());
+        assertEquals(NOT_FOUND.getStatusCode(), t2.status());
+    }
+
+    @Test
+    public void testConcurrentPostDatastreamResource() throws IOException, InterruptedException {
+        final String pid = getRandomUniqueId();
+        createObject(pid);
+
+        final RequestThread t1 = new RequestThread(postDSFileMethod(pid, TEST_FILE));
+        final RequestThread t2 = new RequestThread(postDSFileMethod(pid, TEST_FILE));
+        t1.start();
+        Thread.sleep(100);
```

Comment:
@ajs6f I think this is just imitate the behavior of the HTML UI and make sure that first thread is done with status code 201 while the second thread failed with status code 409 or others. Thanks.